Assets/BowAndArrowImproved/Scripts/Leaderboard/Leaderboard.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using UnityEngine;
+
+public class Leaderboard : MonoBehaviour
+{
+    private const string JsonLeaderBoardFileName = "JsonLeaderboardHolder.json";
+    private LeaderboardEntries _leaderboardEntries;
+
+    private IDataLoader _leaderboardLoader = new JsonDataLoader();
+    private IDataSaver _leaderboardSaver = new JsonDataSaver();
+
+    [SerializeField] private Scoreboard scoreboard;
+    [SerializeField] private GameObject scoreEntriesGrid;
+    [SerializeField] private ScoreEntry scoreEntryPrefab;
+    [SerializeField] private int numOfEntries = 10;
+
+    private List<ScoreEntry> _scoresEntries = new List<ScoreEntry>();
+
+    private int _myScoreEntryPosition = -1;
+    private int _myScore = 0;
+
+    public void OnGameManagerStateChanged(GameManager.State newState)
+    {
+        if (newState == GameManager.State.Lose)
+        {
+            LoadLeaderboardData();
+
+            _myScore = scoreboard._score;
+            SubmitScore();
+
+            GenerateLeaderboardEntries();
+            gameObject.SetActive(true);
+        }
+    }
+
+    private void LoadLeaderboardData()
+    {
+        string filePath = Path.Combine(Application.persistentDataPath, JsonLeaderBoardFileName);
+
+        if (!File.Exists(filePath))
+        {
+            LeaderboardEntries leaderboardData = new LeaderboardEntries
+            {
+                leaderboard = new List<SingleLeaderboardEntry>()
+            };
+
+            string jsonDataToWrite = JsonUtility.ToJson(leaderboardData);
+
+            File.WriteAllText(filePath, jsonDataToWrite);
```

Comment:
Extract this to a different function called `CreateLeaderboardDataFile()` or something like that